### ISSUE TITLE
Summary not working with Model made out of submodules

### ISSUE THREAD

**Issue by RHTNT** [TylerYep/torchinfo#313]
**Describe the bug**
The model summary is incomplete/wrong and shows no parameters, see the attached screenshot.

**To Reproduce**
`def GetConvBlockOutputDim(conv_block, H_in, W_in):
    conv = conv_block.conv
    pool = conv_block.pool

    padding = conv.padding
    stride = conv.stride
    kernel_size = conv.kernel_size

    pool_kernel_size = pool.kernel_size
    pool_stride = pool.stride
    pool_padding = pool.padding

    ComputeOutputDim = lambda in_dim, k, p, s: int(np.floor((in_dim - k + 2 * p) / s) + 1)

    # conv layer
    H_out = ComputeOutputDim(H_in, kernel_size[0], padding[0], stride[0])
    W_out = ComputeOutputDim(W_in, kernel_size[1], padding[1], stride[1])

    # pooling layer
    H_out = ComputeOutputDim(H_out, pool_kernel_size, pool_padding, pool_stride)
    W_out = ComputeOutputDim(W_out, pool_kernel_size, pool_padding, pool_stride)

    return H_out, W_out
`

`class ConvPool2d(nn.Module):
    def __init__(self, in_channels, out_channels, kernel_size, pool_size, conv_stride=1, padding=0):
        super(ConvPool2d, self).__init__()

        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size, stride=conv_stride, padding=padding)

        self.pool = nn.MaxPool2d(pool_size)

    def forward(self, x):
        x = self.conv(x)
        x = F.relu(x)  # Assuming ReLU activation function; you can change or remove it
        x = self.pool(x)
        return x

class FCBlock(nn.Module):
    def __init__(self, input_dim, output_dim, p_dropout, activation=torch.relu):
        super(FCBlock, self).__init__()
        self.fc = nn.Linear(input_dim, output_dim)
        self.activation = activation
        self.p_dropout = p_dropout

    def forward(self, x):
        x = self.fc(x)
        x = self.activation(x)
        x = F.dropout(x, p=self.p_dropout, training=self.training)
        return x

class GenericCNN(nn.Module):
    def __init__(self, input_shape, convblocks_list, fcblocks_list,  final_activation = F.log_softmax):
        super(GenericCNN, self).__init__()
        self.H_in = input_shape[0]
        self.W_in = input_shape[1]
        self.in_channels  = 3

        self.conv_blocks = self.BuildConvBlocks(convblocks_list)
        self.fc_blocks   = self.BuildFCLayers(fcblocks_list)
        self.final_activation = final_activation

    def BuildConvBlocks(self, convblocks_list):
        conv_blocks = []
        for i, convblock_dict in enumerate(convblocks_list):
            if i == 0:
                conv_blocks.append(ConvPool2d(self.in_channels, convblock_dict["out_channels"], convblock_dict["kernel_size"], convblock_dict["pool_size"]))
            else:
                conv_blocks.append(ConvPool2d(convblocks_list[i-1]["out_channels"], convblock_dict["out_channels"], convblock_dict["kernel_size"],
                                              convblock_dict["pool_size"]))
        return conv_blocks

    def GetFinalConvBlockOutputDim(self):
        H = self.H_in
        W = self.W_in

        for conv_block in self.conv_blocks:
            H, W = GetConvBlockOutputDim(conv_block, H, W)

        return H * W * self.conv_blocks[-1].conv.out_channels

    def BuildFCLayers(self, fcblocks_list):
        fc_blocks = []
        conv_output_dim = self.GetFinalConvBlockOutputDim()
        for i, fcblocks_dict in enumerate(fcblocks_list):
            if i == 0:
                fc_blocks.append(
                    FCBlock(conv_output_dim, fcblocks_dict["out_channels"], fcblocks_dict["p_dropout"], fcblocks_dict["activation"]))
            else:
                fc_blocks.append(FCBlock(fcblocks_list[i-1]["out_channels"], fcblocks_dict["out_channels"], fcblocks_dict["p_dropout"], fcblocks_dict["activation"]))

        return fc_blocks

    def forward(self,x):
        for conv_block in self.conv_blocks:
            x = conv_block(x)

        x = torch.flatten(x,1)

        for fc_block in self.fc_blocks:
            x = fc_block(x)

        return self.final_activation(x, dim=1)`

Summary call: 
def GenerateRandomConvBlock():
    convblock = {"out_channels":random.randint(1,64), "kernel_size":random.randint(1,7), "pool_size":random.randint(1, 4)}
    return convblock

def GenerateRandomFCBlock():
    activation_list = [nn.Sigmoid(), nn.ReLU(), nn.Identity(), nn.SELU()]
    fcblock = {"in_channels":random.randint(1,100), "out_channels":random.randint(1,100), "p_dropout":random.uniform(0,0.6), "activation":random.choice(activation_list)}
    return fcblock

img_shape = (162, 402)
batch_size = 64

input_shape = (3, *img_shape)
for i in range(0,20):
    convblock_list = []
    fcblock_list = []

    for k in range(random.randint(1,3)):
        convblock_list.append(GenerateRandomConvBlock())
    for s in range(random.randint(1,3)):
        fcblock_list.append(GenerateRandomFCBlock())

    model = GenericCNN(img_shape, convblock_list, fcblock_list).cpu()

    summary(model, input_size=(batch_size, *input_shape), device='cpu')

**Expected behavior**
I expected the output to show the individual layers, their shapes and the parameter count. I debugged and the layers are set up correctly with correct number of weights

**Screenshots**
This is what I get:
![summary](https://github.com/TylerYep/torchinfo/assets/77005297/dd33de11-fcfe-4cd8-b2d2-5fce9f7b0422)

**Desktop (please complete the following information):**
I run the code on OpenSuse using Pycharm.

**Additional context**
I tried to build a generic CNN, which appears to build correctly going by my debug information. I wanted to use torchinfo to double check the correctness of my code, which lead me to this apparent bug.

**Comment by RHTNT** [comment]
This behavior is due to not calling nn.ModuleList to register the layer blocks, e.g., one has to write 
        `self.conv_blocks = nn.ModuleList(self.BuildConvBlocks(convblocks_list))`
instead of 
`self.conv_blocks =self.BuildConvBlocks(convblocks_list)`
